packages/opencensus-core/src/trace/model/tracer.ts

```diff
@@ -81,6 +88,28 @@ export class CoreTracer implements types.Tracer {
     this.config = config;
     this.logger = this.config.logger || logger.logger();
     this.sampler = SamplerBuilder.getSampler(config.samplingRate);
+    if (config.traceParameters != null) {
+      this.numberOfAnnontationEventsPerSpan =
```

Comment:
What would you think about integrating the default values right in line here rather than abstracting it into a builder class?

It could just be something like:
```
this.traceParameters.numberOfAnnontationEventsPerSpan = 
    config.traceParameters.numberOfAnnontationEventsPerSpan ||
        DEFAULT_NUMBER_OF_ANNOTATION_EVENTS_PER_SPAN
```

and similar for the other parameters.

I personally think the builder pattern, while it makes sense for other languages is not needed in TypeScript.

Reply:
I agree with @draffensperger, builder pattern is not required here.

Reply:
https://github.com/census-instrumentation/opencensus-node/pull/311#discussion_r252286902